driver/src/main/java/org/neo4j/driver/TransactionConfig.java

```diff
@@ -189,25 +189,38 @@ private Builder()
 
         /**
          * Set the transaction timeout. Transactions that execute longer than the configured timeout will be terminated by the database.
+         * See also {@link #withDefaultTimeout}.
          * <p>
          * This functionality allows to limit query/transaction execution time. Specified timeout overrides the default timeout configured in the database
          * using {@code dbms.transaction.timeout} setting.
          * <p>
-         * Provided value should not be {@code null} and should not represent a duration of zero or negative duration.
+         * Provided value should not represent a negative duration.
          *
          * @param timeout the timeout.
          * @return this builder.
          */
         public Builder withTimeout( Duration timeout )
         {
             requireNonNull( timeout, "Transaction timeout should not be null" );
-            checkArgument( !timeout.isZero(), "Transaction timeout should not be zero" );
             checkArgument( !timeout.isNegative(), "Transaction timeout should not be negative" );
 
             this.timeout = timeout;
             return this;
         }
 
+        /**
+         * Set the transaction timeout to ths server-side configured default timeout. This is the default behaviour if neiter
+         * {@link #withTimeout} has not been called.
```

Comment:
Please check wording here. There also seems to be typo in front of `server-side` part.